.github/workflows/ci.yml

```diff
@@ -28,6 +28,17 @@ permissions:
   contents: read
 
 jobs:
+  # Free disk space from the github ubuntu image, as is very bloated (Android and DotNet are safe and about ~11G)
+  # Task is overlaped with other runs but should not cause a problem
```

Comment:
instead I would add `needs: free-disk`  to ensure that this job is run before the others
otherwise if for whatever reason this one does not start, then the others might randomly fail

see https://docs.github.com/en/actions/using-workflows/about-workflows#creating-dependent-jobs

Reply:
Sadly is more complex...

It appears that every job runs in an independent worker ( which is tied to a different `ubuntu-latest` image) which means that changes in the worker do not propagate to other jobs, being dependents or not.

So basically when I run this:
1 - Workflow starts
2 - free-disk and ubuntu-fulldbg-gcc runs, each in a different worker
3 - free-disk ends and its worker ends with it, so no changes in disk

With the `needs`:
1 - Workflow starts
2 - free-disk starts, with its assigned worker, ubuntu-fulldbg-gcc gets into queue
3 - free-disk ends and its worker ends with it. ubuntu-fulldbg-gcc starts with another worker.
4 - We end up in the same place.

By manually deleting those libs the problem its solved, but there is a detail which is preventing it to work. The rationale behind separating the free-disk step into a different job, is that compilation jobs are running inside custom containers, so I don't have direct access to the worker.

As the jobs do not share worker, and the jobs that need more space run in a custom container, it is impossible to liberate space normally.

At this points I am trying several solutions:
1 - Trim the KratosUbuntu-22.04 image with less software ( difficult, as the main offender here is Intel's OneApi 8G, 90% of the image size, and we need it )
2 - Divide the compilation in substeps and clear the intermediate compilation artifacts
3 - Manually mount the docker image and create a custom entry point that executes. Feasible but a tone of work and will prevent us from splinting the job in different steps.
4 - Mounting the host runner directories that we want to delete in the container, and delete its contents. ( God forgive me for suggesting this)
5 - Pray for github to implement pre-step and post-step. (Spoiler: not gonna happen)

I am trying to implement 2, but if not the case we will have to think the whole process differently....

Reply:
Ah so it is kinda a coincidence that the current solution allows to run the free-disk job on the same worker?

Regarding your options, I already optimized the image quite a bit regarding size

Also I tried to use custom entry points some time ago and it was not (yet?) supported

So if it works, then kudos for figuring it out

Reply:
We could split the image into two, one base image with all the compilers and one with the intel stuff 
The intel build only has the core atm, which takes less space

Only downside is that the gcc builds would not have Pardiso enabled

Reply:
Also there are multiple versions of mmg and parmmg installed

@loumalouomega @marcnunezc can we remove the older versions to safe space?

Reply:
Hi!

Currently, the unused versions are (which can be safely removed):

https://github.com/KratosMultiphysics/Kratos/blob/f72bc3d9ef01dd35a2ac2dbf6bc8a799357e4239/scripts/docker_files/docker_file_ci_ubuntu_22_04/DockerFile#L67-L74

https://github.com/KratosMultiphysics/Kratos/blob/f72bc3d9ef01dd35a2ac2dbf6bc8a799357e4239/scripts/docker_files/docker_file_ci_ubuntu_22_04/DockerFile#L85-L94

These were part of an upgrade I did not finish in https://github.com/KratosMultiphysics/Kratos/pull/10477 since some tests were giving problems. Also, a new version of MMG was released so it will be better to start all over again with the more recent releases.

Reply:
unused MMG versions removed in #11587